src/components/controller.cpp

```diff
@@ -386,36 +390,195 @@ void controller::process_eventqueue() {
   }
 }
 
+/**
+ * Tries to match the given command to a legacy action string and sends the
+ * appropriate new action (and data) to the right module if possible.
+ *
+ * \returns true iff the given command matches a legacy action string and was
+ *          successfully forwarded to a module
+ */
+bool controller::try_forward_legacy_action(const string cmd) {
+  /*
+   * Maps legacy action names to a module type and the new action name in that module.
+   *
+   * We try to match the old action name as a prefix, and everything after it will also be added to the end of the new
+   * action string (for example "mpdseek+5" will be redirected to "seek.+5" in the first mpd module).
+   *
+   * The action will be delivered to the first module of that type so that it is consistent with existing behavior.
+   * If the module does not support the action or no matching module is found, the command is forwarded to the shell.
+   *
+   * TODO Remove when deprecated action names are removed
+   */
+// clang-format off
+#define A_MAP(old, module_name, event) {old, {string(module_name::TYPE), string(module_name::event)}}
+
+  const std::map<string, std::pair<string, const string>> legacy_actions{
```

Comment:
You could add a `static` to avoid recreation of this map at each call.

Reply:
Good idea

Reply:
Moreover, it can be an `unordered_map` since string are hash-able.